logmatic/hook.go

```diff
@@ -0,0 +1,52 @@
+package logmatic
+
+import (
+	"time"
+
+	logmatic "github.com/logmatic/logmatic-go"
+	"github.com/sirupsen/logrus"
+)
+
+type Config struct {
+	APIKey     string
+	Interval   time.Duration
+	BufferSize int
+}
+
+func NewLogrusHook(config Config) logrus.Hook {
+	if config.APIKey == "" {
+		panic("logmatic API key should is required")
```

Comment:
"should is" and IMO this lib should return an error and let the client decide if it want's to raise a panic.

Reply:
oops! typo

Reply:
About the `panic`, here is an excerpt from [Effective Go's Panic section](https://golang.org/doc/effective_go.html#panic):

> One possible counterexample is during initialization: if the library truly cannot set itself up, it might be reasonable to panic, so to speak.
> 
> ```go
> var user = os.Getenv("USER")
> 
> func init() {
>     if user == "" {
>         panic("no value for $USER")
>     }
> }